test/use_text_editing_controller_test.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/src/framework.dart';
+import 'package:flutter_hooks/src/hooks.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'mock.dart';
+
+void main() {
+  testWidgets('useTextEditingController returns a controller', (tester) async {
+    TextEditingController controller;
+
+    await tester.pumpWidget(HookBuilder(
+      builder: (context) {
+        controller = useTextEditingController();
+        return Container();
+      },
+    ));
+
+    expect(controller, isNotNull);
+    controller.addListener(() {});
```

Comment:
Thinking about that, should the hook `dispose` contains an assert that checks that the controller has no more listeners (using `controller.hasListeners`)?

Reply:
I'm not sure about that - I think there are valid use cases where a widget (or a builder in a subtree) listens to a text controller without calling `removeListener`. Since the widget manages the controller and all listeners, the developer can omit the call to `removeListener` without leaking resources.

After all, `_AnimationControllerHookState` doesn't do a similar check either.

Reply:
Okey. If it caused a problem, it's the job of ValueNotifier/ChangeNotifier to make such assert anyway